Marlin/src/gcode/gcode.cpp

```diff
@@ -171,19 +181,65 @@ void GcodeSuite::get_destination_from_command() {
     constexpr bool skip_move = false;
   #endif
 
+  #if ANY(ROTATE_WORKSPACE, SCALE_WORKSPACE)
+    static xyz_pos_t raw_destination; // {0}
+  #endif
+
   // Get new XYZ position, whether absolute or relative
   LOOP_NUM_AXES(i) {
     if ( (seen[i] = parser.seenval(AXIS_CHAR(i))) ) {
-      const float v = parser.value_axis_units((AxisEnum)i);
-      if (skip_move)
+      if (skip_move) {
+        #if ANY(SCALE_WORKSPACE, ROTATE_WORKSPACE)
+          raw_destination[i] = current_position[i];
+        #else
+          destination[i] = current_position[i];
+        #endif
+      }
+      else {
+        const float v = parser.value_axis_units((AxisEnum)i);
+        #if ANY(SCALE_WORKSPACE, ROTATE_WORKSPACE)
+          raw_destination[i] = axis_is_relative(AxisEnum(i)) ? raw_destination[i] + v : LOGICAL_TO_NATIVE(v, i);
+        #else
+          destination[i] = axis_is_relative(AxisEnum(i)) ? current_position[i] + v : LOGICAL_TO_NATIVE(v, i);
+        #endif
+      }
+    }
+    else {
+      #if ANY(SCALE_WORKSPACE, ROTATE_WORKSPACE)
+        raw_destination[i] = current_position[i];
+      #else
         destination[i] = current_position[i];
-      else
-        destination[i] = axis_is_relative(AxisEnum(i)) ? current_position[i] + v : LOGICAL_TO_NATIVE(v, i);
+      #endif
     }
-    else
-      destination[i] = current_position[i];
   }
 
+  #if ANY(SCALE_WORKSPACE, ROTATE_WORKSPACE)
+    destination = raw_destination;
+  #endif
+
+  #if ENABLED(SCALE_WORKSPACE)
+    if (!(NEAR(scaling_factor.x, 1.0f) || NEAR(scaling_factor.y, 1.0f) || NEAR(scaling_factor.z, 1.0f))) {
```

Comment:
Shouldn't this be: `if (!(NEAR(scaling_factor.x, 1.0f) && NEAR(scaling_factor.y, 1.0f) && NEAR(scaling_factor.z, 1.0f)))`?

Frankly, there will never be any case where a scaling factor is 0.99999 or 1.000001 when it was explicitly set to 1.0 by any method. So it should be absolutely fine to use `if (scaling_factor.x != 1.0f || scaling_factor.y != 1.0f || scaling_factor.z != 1.0f)`.